src/OpenTelemetry.Instrumentation.AspNetCore/TracerProviderBuilderExtensions.cs

```diff
@@ -53,8 +53,19 @@ public static class TracerProviderBuilderExtensions
             this TracerProviderBuilder builder,
             AspNetCoreInstrumentation instrumentation)
         {
-            builder.AddSource(HttpInListener.ActivitySourceName);
-            builder.AddLegacySource(HttpInListener.ActivityOperationName); // for the activities created by AspNetCore
+            // For .NET7.0 onwards activity will be created using activitySource.
+            // https://github.com/dotnet/aspnetcore/blob/main/src/Hosting/Hosting/src/Internal/HostingApplicationDiagnostics.cs#L327
+            // For .NET6.0 and below, we will continue to use legacy way.
+            if (HttpInListener.IsNet7OrGreater)
+            {
+                builder.AddSource(HttpInListener.FrameworkActivitySourceName);
```

Comment:
There is a subtle potential bug/issue with how this is being done which may or may not be a concern. If we consider what [AspNetCore is doing](https://github.com/dotnet/aspnetcore/blob/main/src/Hosting/Hosting/src/WebHostBuilder.cs#L291)...

```csharp
services.TryAddSingleton(sp => new ActivitySource("Microsoft.AspNetCore"));
```

...that is not a fixed/set in stone mechanism. It is a _try_. Users may register their own activity source before that one, or they may change it after.

Maybe we shouldn't access by name at all and instead do something like...

```csharp
var activitySource = _applicationServices.GetRequiredService<ActivitySource>();
```

...to access the runtime registered `ActivitySource` which is what AspNetCore itself is doing: https://github.com/dotnet/aspnetcore/blob/0e5cbf447d329a1e7d69932c3decd1c70a00fbba/src/Hosting/Hosting/src/Internal/WebHost.cs#L152